doc/developer/101-query-compilation.md

```diff
@@ -34,7 +41,16 @@ Transformations in the compile-time lifecycle of a SQL statement.
         * Machinery for introducing errors for `SELECT` subqueries with more than one return value.
     * `EXPLAIN DECORRELATED` returns the result of transformations up to this point.
 * [`MIR ⇒ MIR`](https://github.com/MaterializeInc/materialize/blob/main/src/transform).
-    * Optimizations:
+    * [If the query is a view
+      definition](https://github.com/MaterializeInc/materialize/blob/main/src/coord/src/catalog.rs#L3325),
+      run per-view logical optimizations against the SQL query. The catalog
+      stores the result of transformations up to this point.
+    * [Construct a dataflow for the query](https://github.com/MaterializeInc/materialize/blob/main/src/coord/src/coord/dataflow_builder.rs):
+        * If the query depends on not-materialized views, the definitions of the
+          not-materialized views get inlined.
+        * Import all indexes belonging to a materialized view that the query
```

Comment:
I'm not sure when an index is considered to "belong to a materialized view". Is it when it is listed in the `index_exports` field of the `DataflowDescription` object for the materialized view?